sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/testing/TestDataflowPipelineRunner.java

```diff
@@ -220,10 +220,11 @@ public String toString() {
   private static class CancelWorkflowOnError implements JobMessagesHandler {
     private final DataflowPipelineJob job;
     private final JobMessagesHandler messageHandler;
-    private volatile String errorMessage = "";
+    private final StringBuffer errorMessage;
```

Comment:
True, but the presence of volatile makes me think there's some concurrency here, and StringBuffer is thread-safe